build.sbt

```diff
@@ -0,0 +1,183 @@
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain
+
+val commonSettings = Seq(
+  organization := "com.mesosphere.usi",
+  version := {
+    import sys.process._
+      ("./version" !!).trim
+  },
+  libraryDependencies := Seq(
+    Dependencies.Test.akkaSlf4j % "test",
+    Dependencies.Test.logbackClassic % "test"
+  ),
+  // the PortAllocator logic is not cross-module aware, so running concurrent tests causes failures
+  concurrentRestrictions in Global += Tags.limit(Tags.Test, 1),
+  scalaVersion := "2.13.1",
+  crossScalaVersions := Seq("2.13.1", "2.12.11"),
+  addCompilerPlugin(scalafixSemanticdb),
+  scalacOptions ++= List(
+    "-Yrangepos",
+    "-Ywarn-unused"
+  ),
+  publishTo := Some(s3resolver.value(
+    "Mesosphere Public Repo (S3)",
+    s3("downloads.mesosphere.io/maven")
+  )),
+  s3credentials := DefaultAWSCredentialsProviderChain.getInstance(),
+  s3region :=  com.amazonaws.services.s3.model.Region.US_Standard,
+)
+
+lazy val `root` = (project in file("./"))
+  .settings(commonSettings)
+  .settings(publish / skip := true)
+  .aggregate(
+    `core-models`,
+    `metrics`,
+    `persistence`,
+    `test-utils`,
+    `mesos-client`,
+    `persistence-zookeeper`,
+    `metrics-dropwizard`,
+    `mesos-master-detector`,
+    `core`,
+    `examples-core-hello-world`,
+    `examples-keep-alive-framework`,
+    `examples-mesos-client-example`,
+    `examples-simple-hello-world`)
+
+lazy val `core-models` = (project in file("./core-models/"))
+  .settings(commonSettings)
+  .settings(
+    libraryDependencies ++= Seq(
+      Dependencies.mesos,
+      Dependencies.scalaLogging,
+      Dependencies.Test.scalaTest % "test")) // note, core-models cannot depends on test-utils because this creates a circular dependency.
+
+lazy val `metrics` = (project in file("./metrics"))
+  .settings(commonSettings)
+  .settings(
+    libraryDependencies ++= Seq(
+      Dependencies.akkaStream))
+
+lazy val `persistence` = (project in file("./persistence"))
+  .settings(commonSettings)
+  .settings(
+    libraryDependencies ++= Seq(
+      Dependencies.akkaActor
+    ))
+  .dependsOn(`core-models`)
+
+lazy val `test-utils` = (project in file("./test-utils/"))
+  .settings(commonSettings)
+  .settings(
+    libraryDependencies ++= Seq(
+      Dependencies.Test.scalaTest,
+      Dependencies.akkaStream,
+      Dependencies.scalaAsync % "compile",
+      Dependencies.Test.junit,
```

Comment:
Aren't we missing `% test` here as well?

Reply:
Actually, I decided against doing the `% "test"` etc here because test-utils includes test dependencies in main. So it's up to each project what gets included where.